alea/utils.py

```diff
@@ -215,3 +327,222 @@ def clip_limits(value) -> Tuple[float, float]:
+    variations = evaluate_numpy_scipy_expression_in_dict(variations)
+
+    # expand to grid if necessary
+    for k, v in variations.items():
+        if not isinstance(v, list):
+            raise ValueError(f"variations {k} must be a list, not {v} with {type(v)}")
+        variations[k] = expand_grid_dict(v)
+    result = [dict(zip(variations, t)) for t in iteration(*variations.values())]
+    if result:
+        return result
+    else:
+        return [{}]
+
+
+def convert_to_zip(to_zip: Dict[str, List]) -> List[Dict[str, Any]]:
+    """Convert dict into a list of dict, according to the zip method.
+
+    Example:
+        >>> convert_to_zip({'a': [1, 2], 'b': [3, 4]})
+        [{'a': 1, 'b': 3}, {'a': 2, 'b': 4}]
+
+    """
+    return convert_variations(to_zip, zip)
+
+
+def convert_to_vary(to_vary: Dict[str, List]) -> List[Dict[str, Any]]:
+    """Convert dict into a list of dict, according to the itertools.product method.
+
+    Example:
+        >>> convert_to_vary({'a': [1, 2], 'b': [3, 4]})
+        [{'a': 1, 'b': 3}, {'a': 1, 'b': 4}, {'a': 2, 'b': 3}, {'a': 2, 'b': 4}]
+
+    """
+    return convert_variations(to_vary, itertools.product)
+
+
+def convert_to_in_common(in_common: Dict[str, Any]) -> Dict[str, Any]:
+    """Expand the values in in_common, according to the itertools.product method, if necessary. This
+    usually happens to the hypotheses.
+
+    Example:
+        >>> convert_to_in_common({'hypotheses': ['free', {'a': [1, 2], 'b': [3, 4]}]})
+        {
+            "hypotheses": [
+                "free",
+                {"a": 1, "b": 3},
+                {"a": 1, "b": 4},
+                {"a": 2, "b": 3},
+                {"a": 2, "b": 4},
+            ]
+        }
+
+    """
+    for k, v in in_common.items():
+        if isinstance(v, list) and (k != "hypotheses") and (k != "confidence_levels"):
+            raise ValueError(
+                f"except hypotheses, in_common can not contain list, "
+                f"you might need to put {(k, v)} in to_zip or to_vary"
+            )
+    if "hypotheses" in in_common:
+        in_common["hypotheses"] = expand_grid_dict(in_common["hypotheses"])
+    return in_common
+
+
+def compute_variations(to_zip, to_vary, in_common) -> list:
+    """Compute variations of Runner from to_zip, to_vary and in_common. By priority, the order is
+    to_zip, to_vary, in_common. The values in to_zip will overwrite the keys in to_vary and
+    in_common. The values in to_vary will overwrite the keys in in_common.
+
+    Args:
+        to_zip (dict): variations to be zipped
+        to_vary (dict): variations to be varied
+        in_common (dict): variations in common
+
+    Returns:
+        list: a list of dict
+
+    """
+    zipped = convert_to_zip(to_zip=to_zip)
+    varied = convert_to_vary(to_vary=to_vary)
+
+    combined = [
+        {**convert_to_in_common(in_common), **v, **z} for z, v in itertools.product(zipped, varied)
+    ]
+    return combined
+
+
+def hashablize(obj):
```

Comment:
I really like this new verb 😄 But I guess `make_hashable` is a bit more intuitive.

Reply:
Changed.